src/EventProducerService.cpp

```diff
@@ -23,173 +23,227 @@ void EventProducerService::setController(Controller *cntrl)
+void EventProducerService::createDefault(byte evValue)
+{
+  byte data[4];
+  byte index = module_config->findEventSpace();
+  data[0] = highByte(module_config->nodeNum);
+  data[1] = lowByte(module_config->nodeNum);
+  data[2] = 0;
+  data[3] = evValue;
+  
+  module_config->writeEvent(index, data);
+  module_config->writeEventEV(index, 1, evValue);
+  module_config->updateEvHashEntry(index);   
+}
+
+void EventProducerService::sendEvent(bool state, byte evValue)
 {
   byte nn_en[4];
-  module_config->readEvent(index, nn_en);
+  byte index;
   byte opCode;
-  unsigned int nn = ((nn_en[0] << 8) && nn_en[1]);
-  if (nn == 0)
+  index = module_config->findExistingEventByEv(1, evValue);
+  if (index < module_config->EE_MAX_EVENTS)
   {
-    opCode = (state ? OPC_ASON : OPC_ASOF);
-    nn_en[0] = highByte(module_config->nodeNum);
-    nn_en[1] = lowByte(module_config->nodeNum); 
-  }
-  else
-  {
-    opCode = (state ? OPC_ACON : OPC_ACOF);
-  }
-  
-  VlcbMessage msg;
-  msg.len = 5;
-  msg.data[0] = opCode;
-  msg.data[1] = nn_en[0];
-  msg.data[2] = nn_en[1];
-  msg.data[3] = nn_en[2];
-  msg.data[4] = nn_en[3];
-  controller->sendMessage(&msg);
+    module_config->readEvent(index, nn_en);
+    unsigned int nn = ((nn_en[0] << 8) && nn_en[1]);
+    if (nn == 0)
+    {
+      opCode = (state ? OPC_ASON : OPC_ASOF);
+      nn_en[0] = highByte(module_config->nodeNum);
+      nn_en[1] = lowByte(module_config->nodeNum); 
+    }
+    else
+    {
+      opCode = (state ? OPC_ACON : OPC_ACOF);
+    }
     
-  if (coeService)
+    VlcbMessage msg;
+    msg.len = 5;
+    msg.data[0] = opCode;
+    msg.data[1] = nn_en[0];
+    msg.data[2] = nn_en[1];
+    msg.data[3] = nn_en[2];
+    msg.data[4] = nn_en[3];
+    controller->sendMessage(&msg);
+      
+    if (coeService)
+    {
+      coeService->put(&msg);
+    }
+  }
+  else  // Produced event doesn't exist so create a default
   {
-    coeService->put(&msg);
+    createDefault(evValue);
   }
 }
 
-void EventProducerService::sendEvent(bool state, byte index, byte data1)
+void EventProducerService::sendEvent(bool state, byte evValue, byte data1)
 {
   byte nn_en[4];
-  module_config->readEvent(index, nn_en);
+  byte index;
   byte opCode;
-  
-  if ((nn_en[0] == 0) && (nn_en[1] == 0))
-  {
-    opCode = (state ? OPC_ASON1 : OPC_ASOF1);
-    nn_en[0] = highByte(module_config->nodeNum);
-    nn_en[1] = lowByte(module_config->nodeNum); 
-  }
-  else
+  index = module_config->findExistingEventByEv(1, evValue);
+  if (index < module_config->EE_MAX_EVENTS)
   {
-    opCode = (state ? OPC_ACON1 : OPC_ACOF1);
+    module_config->readEvent(index, nn_en);
+    unsigned int nn = ((nn_en[0] << 8) && nn_en[1]);
+    if (nn == 0)
+    {
+      opCode = (state ? OPC_ASON1 : OPC_ASOF1);
+      nn_en[0] = highByte(module_config->nodeNum);
+      nn_en[1] = lowByte(module_config->nodeNum); 
+    }
+    else
+    {
+      opCode = (state ? OPC_ACON1 : OPC_ACOF1);
+    }
+    
+    VlcbMessage msg;
+    msg.len = 5;
```

Comment:
No, this should still be 6.

Reply:
Your right.  My mistake.  Line 184 should be 7, as well.